import/popups/DownloadMenuItem.qml

```diff
@@ -36,7 +37,7 @@ MenuItem {
         return DownloadHelper.createUniqueFileUrl(leafName, targetDirectory)
     }
 
-    onClicked: {
+    function startDownload() {
```

Comment:
It looks to me like you'd now be better off dropping `downloadFileTargetUrl` entirely as a property and evaluating it here instead. You'd need to remove it from ContextMenu.qml too (which I think is also the correct thing to do).